src/Context/ContextHandlerIntegrityTrait.php

```diff
@@ -131,28 +131,12 @@ protected function doCheckIntegrity(CoreContextAwarePluginInterface $plugin, Exe
    * @param \Drupal\rules\Engine\IntegrityViolationList $violation_list
    *   The list of violations where new ones will be added.
    */
-  protected function checkDataTypeCompatible(ContextDefinitionInterface $context_definition, DataDefinitionInterface $provided, $context_name, IntegrityViolationList $violation_list) {
-    $expected_class = $context_definition->getDataDefinition()->getClass();
-    $provided_class = $provided->getClass();
```

Comment:
I made type checking strict until we have a more flexible option. This was necessary for the test case to pass; i.e. to fail integrity check as expected.